perses/annihilation/ncmc_switching.py

```diff
@@ -135,6 +143,30 @@ def _getAvailableParameters(self, system, prefix='lambda'):
                         parameters.append(parameter_name)
         return parameters
 
+    def _updateAlchemicalState(self, context, functions, value):
+        """
+        Update alchemical state using the specified lambda value.
+
+        Parameters
+        ----------
+        context : simtk.openmm.Context
+            The Context
+        functions : dict
+            A dictionary of functions
+        value : float
+            The alchemical lambda value
+
+        TODO: Improve function evaluation to better match Lepton and be more flexible in exact replacement of 'lambda' tokens
+        Perhaps this could be done with a special integrator?
+
+        """
+        from perses.annihilation import NumericStringParser
```

Comment:
Ok, if this is not ridiculous, it's probably preferable to do this inside an integrator?